### PR TITLE
improve doc on getFieldState

### PR DESCRIPTION
**Pull request by bluebill1049**
close issue: https://github.com/react-hook-form/react-hook-form/issues/8786#issuecomment-1204570567

### CODE REVIEW

File: src/components/useForm/GetFieldState.tsx [hunk 1]
```diff
@@ -175,18 +188,36 @@ getFieldState('non-existent-name'); // ❌ will return state as false and error
                 />
               </li>
               <li>
-                <p>formState will need to be subscribed.</p>
+                <p>
+                  <code>formState</code> from <code>useForm</code> will need to
```
Comment (fzaninotto) [inline src/components/useForm/GetFieldState.tsx:192]: I don't understand this sentence. Do you mean

> `getFieldState` works by "subscribing" to the form state. That's why it requires the form state as second argument.
> ```jsx
> const { error } = getFieldState('firstName', methods.formState)
> ```
>
> Other hooks (`useForm` or `useFormState`) also "subscribe" to the form state. For instance, this call subscribes to the `errors` part of the form state:
> ```jsx
> const { formState: { errors} } = useForm() // errors is reactive because useForm placed a subscription on it
> ```
> If such a subscription already exists, you can omit the second parameter to `getFieldState`, as it will take advantage of the existing subscription. 
> ```jsx
> const { error } = getFieldState('firstName');
> ```

Writing this, I realize that there may be a correspondence between the form subscriptions (e.g. `errors`)  and the field subscriptions (e.g. `error`). You should probably explain that correspondence somewhere.

File: src/components/useForm/GetFieldState.tsx [hunk 2]
```diff
@@ -175,18 +188,36 @@ getFieldState('non-existent-name'); // ❌ will return state as false and error
...
                 <p>You can subscribe to the formState in two different ways:</p>
 
                 <ol>
                   <li>
                     <p>
-                      You can simply define the required state information when
-                      destructuring the returned value of a relevant hook, such
-                      as <code>useForm</code> (at global level) or{" "}
-                      <code>useFormState</code> (at hook level).
+                      You can define form state subscription at the{" "}
```
Comment (fzaninotto) [inline src/components/useForm/GetFieldState.tsx:203]: This is really hard to follow. I really wonder if it is necessary to explain this use case? Why not only document the use case where the developer has to pass the form state as second parameter?

**Comment by bluebill1049** [comment]
thanks for the comments @fzaninotto they are all valid, I will simplify the doc a bit so it's not misleading the developers. The key information here is `getFieldState` is just a function that needs a `formState` (dirty/errors/touched) subscription before it works properly.
